### DISCUSSION TITLE
Team B - Course 3

### DISCUSSION THREAD

**Discussion by mindgitrwx** [lablup/contribution-academy-2021#60] [Show and tell]
.

**Comment by TeusEE** [comment]
1. backend.ai run -c 'print("hello world!")' python 출력 결과
![129981374-977fa4cb-1d46-4a9d-a31e-e55a9b5c2951](https://user-images.githubusercontent.com/79349894/130157862-9663fd85-d223-4350-882f-f1f698a3c0a0.jpg)


2. backend.ai run으로 fashion mnist 실행
tensorflow2.4 기준으로 image pulling 후 실행하였습니다.
![2](https://user-images.githubusercontent.com/79349894/130157871-de6ca96f-fa87-403b-a787-6957c529006f.jpg)

3. webui 환경 구축
![image](https://user-images.githubusercontent.com/79349894/130355416-8b0a4b98-fc63-4749-ba8d-a79331000aef.png)

**Reply by eunjungchoi** [reply]
혹시  tensorflow2.4 기준으로 image pulling  하신 건 어디에서 어떻게 하신건가요?  그 부분에서 막혀있어서요.

**Reply by TeusEE** [reply]
backend.ai admin으로 tensorflow2.4를 alias설정한 다음에 실행했습ㄴ디ㅏ.
문제가 Pulling하는데 1시간 정도 소요되었네요

**Comment by mindgitrwx** [comment]
Executed each component with debug mode.

![manager](https://user-images.githubusercontent.com/37054916/130332271-4ec3c52d-ca6d-48d6-8ca8-098673c9d68f.png)
![storageProxy](https://user-images.githubusercontent.com/37054916/130332275-fa674d9f-ccad-4045-a917-6190b12647af.png)
![webUI](https://user-images.githubusercontent.com/37054916/130332276-46677946-5010-473c-8775-9c057ff66a3c.png)

1. backend.ai run -c 'print("hello world!")' python execution. 
![helloworld](https://user-images.githubusercontent.com/37054916/130332270-b5efe78e-b808-451d-9c0e-35062e978602.png)

2. backend.ai run and fashion mnist execution (failed!)

![session](https://user-images.githubusercontent.com/37054916/130332349-504fe3b2-d02c-42f8-a164-567026f44dea.png)

```
jonghyeon@mindgitrwx:~/backend.ai-dev/client-py$ backend.ai run --rm --exec 'python single_minist.py' tf2 ./single_minist.py ./minist.py
∙ Session name prefix: pysdk-16f0ef3540
✘ [0] BackendAPIError(400, 'Bad Request', {'type': 'https://api.backend.ai/probs/invalid-api-params', 'title': 'Missing or invalid API parameters.', 'msg': 'Unknown image reference: tf2'})
```

**Reply by agatha197** [reply]
pull the tf2 image and try again. 💪

**Reply by eunjungchoi** [reply]
tf2 이미지를 pull 하는 것은 어떻게 하는 것인가요? 저도 같은 문제에 봉착해서요.

**Reply by mindgitrwx** [reply]
@eunjungchoi https://github.com/lablup/contribution-academy-2021/discussions/60#discussioncomment-1218976

**Comment by min1826** [comment]
1. `backend.ai run -c 'print("hello world!")' python` 출력 결과
![image](https://user-images.githubusercontent.com/71456233/130343020-b3d80f03-1d0a-42fe-b27d-d8a34897895d.png)


2. `backend.ai run`으로 fashion mnist 실행
![image](https://user-images.githubusercontent.com/71456233/130361286-eae948ba-f172-40a3-b2c3-5db6adc0bbba.png)

**Comment by AlchemistX** [comment]
# backend.ai run python -c "print('Hello World\!')" 실행 결과
![HelloWorld](https://user-images.githubusercontent.com/1624251/130354158-18c13378-f29f-4823-a859-cee35835050c.png)

# Fashion MNIST 실행 결과
![MNIST 실행](https://user-images.githubusercontent.com/1624251/130354164-0af50d29-9787-414f-abc6-7fdc42db76d0.png)

# Backend.AI-webui 실행 결과
![webui 실행](https://user-images.githubusercontent.com/1624251/130354255-da166b54-85c9-43af-9acf-2ff7436664a0.png)

# Backend.AI 개발 환경 모니터링
![서비스 실행 중](https://user-images.githubusercontent.com/1624251/130354263-38c19cd7-f4e0-4cd9-bca5-24aab158eac8.png)

# Backend.AI-webui 개발 환경 모니터링
![webui 모니터링](https://user-images.githubusercontent.com/1624251/130354326-261ebe10-f4c6-4d56-ab09-b8e52b53f6b0.png)

**Comment by mindgitrwx** [comment]
**Fashion MNIST** 

1. Pull the docker image
`docker pull cr.backend.ai/stable/python-tensorflow:2.4-py38-cuda11.1`

2. Make alias for future use
`backend.ai admin alias-image tf2 cr.backend.ai/testing/python-tensorflow:2.4-py38-cuda11.1`

3. Run sample mnist files.
` backend.ai run --rm --exec 'python single_mnist.py' tf2 ./single_mnist.py ./mnist.py `

![image](https://user-images.githubusercontent.com/37054916/130361008-0868fcb7-a15c-4485-b965-0036fe796f37.png)
